deploy/ansible/worker/include/run_ursula.yml

```diff
@@ -3,22 +3,137 @@
+        verbosity: 0
+        var: geth_dir
+
+    - name: "keystore output"
+      debug:
+        verbosity: 0
+        var: keystore_output
+      when: DECENTRALIZED is undefined or not DECENTRALIZED
+
+    - name: store signer options
+      set_fact:
+        signer_options: "--signer keystore://{{nucypher_container_geth_datadir}}/keystore/{{keystore_output.files[0].path | basename}}"
+      when: DECENTRALIZED is undefined or not DECENTRALIZED
+
+    - name: store empty signer options
+      set_fact:
+        signer_options: ""
+      when: DECENTRALIZED is not undefined and DECENTRALIZED
+
+    - name: "remove known nodes"
+      become: yes
+      file:
+        path: /home/nucypher/nucypher/known_nodes/
+        state: absent
+
+    - name: "get account address from file"
+      become: yes
+      command: 'cat {{geth_dir}}account.txt'
+      register: active_account
+
+
+    # keeping this around for when we merge the datastore PR
+    # - name: "remove database file"
+    #   become: yes
+    #   file:
+    #     path: /home/nucypher/nucypher/ursula.db
+    #     state: absent
+
+    - name: "ensure known nodes certificates directory"
+      become: yes
+      file:
+        path: /home/nucypher/nucypher/known_nodes/certificates
+        state: directory
+
+    - name: "ensure known nodes directory"
+      become: yes
+      file:
+        path: /home/nucypher/nucypher/known_nodes/metadata
+        state: directory
+
+    - name: Find my public ip
+      uri:
+        url: http://ifconfig.me/ip
+        return_content: yes
+      register: ip_response
+
+    - name: "update Ursula worker config"
+      become: yes
+      become_user: nucypher
+      command: "docker run -v /home/nucypher:/root/.local/share/ -e NUCYPHER_KEYRING_PASSWORD -it {{ nucypher_image | default('nucypher/nucypher:latest') }}  nucypher ursula config --provider {{ blockchain_provider }} --worker-address {{active_account.stdout}} --rest-host {{ip_response.content}} --network {{network_name}} {{nucypher_ursula_init_options | default('')}} --config-file /root/.local/share/nucypher/ursula.json"
+      environment:
+        NUCYPHER_KEYRING_PASSWORD: "{{NUCYPHER_KEYRING_PASSWORD}}"
+
+    - name: "Run Staked Ursula (seed node)"
+      become: yes
+      become_user: nucypher
+      when: SEED_NODE_URI is not undefined and inventory_hostname == SEED_NODE_URI
+      docker_container:
+        recreate: yes
+        name: ursula
+        state: started
+        pull: yes
+        log_driver: json-file
+        log_options:
+          max-size: 10m
+          max-file: "5"
+        image: "{{ nucypher_image | default('nucypher/nucypher:latest') }}"
+        restart_policy: "unless-stopped"
+        command: "nucypher ursula run {{nucypher_ursula_run_options | default('')}} {{signer_options}} --lonely --prometheus --metrics-port 9101"
+        volumes:
+          - /home/nucypher:/root/.local/share/
+        ports:
+          - "9151:9151"
+          - "9090:9090"
```

Comment:
If you are exposing port 9090 for prometheus - I don't think it is necessary. Only 9101 is needed for ursula.